skops/hub_utils/_hf_hub.py

```diff
@@ -81,9 +110,10 @@ def _get_example_input(data):
 
     Parameters
     ----------
-    data: array-like
-        The input needs to be either a ``pandas.DataFrame`` or a
-        ``numpy.ndarray``. The first 3 rows are used as example input.
+    data: pandas.DataFrame or array-like
+        The input needs to be anything that can be converted to a 2D
+        ``numpy.ndarray``, including a ``pandas.DataFrame``. The first 3 rows
+        are used as example input.
```

Comment:
we shouldn't just convert to numpy, for instance, if the input is xarray, we can extract column names. The same with cuDF.

Reply:
Do you have a specific list of data types that you'd like to support? Cause there are countless variants of numpy arrays and pandas dataframes, but I don't think we can find a general solution that works for each of them

Reply:
I don't have a list, but I'm not sure if a silent conversion is a good idea. It might be a better idea to fail on containers which we don't understand, and slowly add them as requests come, WDYT?

Reply:
Ok, so if I understood correctly you'd rather fail on, say, cuDF dataframes rather than silently converting them to NumPy arrays, and the reason is because you don't want to risk throwing away information about the column names. Is that right?

Reply:
Yes. And the reason I don't want to miss that information is the following:

We use these feature names to figure out how to pass the information to the model in the inference side. And that API receives data in this format: `{'colA': [list of values], ...}`. If we lose the feature name information here, the user wouldn't know how to send data to the inference backend and the whole inference side of things break. Since saving feature names here serves pretty much no other purpose than making the inference backend, it'd be a pity to silently convert them to numpy arrays and lose that information.

So I think it's better to fail on things we don't understand, and explicitly add the more popular ones. WDYT?

Reply:
Makes total sense to me, however this means we need to explicitly check the type of the inputs we get, right? So, basically, right now we're supporting only:

- `DataFrame`s (for which we can also extract column names)
- `np.array`s
- [list|tuple] of [list|tuple|np.array] (can be silently converted to `np.array`s using sklearn's `check_array`)

Anything else we should add for now?

Reply:
I think this would be a good start. I'm curious to see what people are gonna use. We should ideally raise for all other types or at least raise a warning.